api/v1alpha1/progressiverollout_types.go

```diff
@@ -100,6 +102,34 @@ func (in *ProgressiveRollout) Owns(owners []metav1.OwnerReference) bool {
 	return false
 }
 
+// SetStageStatus sets the corresponding StageStatus in stageStatus to newStatus
+// - If a stage doesn't exist, it will be added to StageStatus slice
+// - If a stage already exists it will be updated
+func (in *ProgressiveRollout) SetStageStatus(newStatus StageStatus, updateTime metav1.Time) {
```

Comment:
so if we're going to leave a pointer for Time, in this instance we should have a pointer parameter too;
that will avoid having to copy `updateTime` as parameter, and then send the pointer back which means the struct will leave the stack and go to the heap.
That will cause some unnecessary garbage collection

Reply:
Alright - I'll change it back - I just asked this in another comment 😛